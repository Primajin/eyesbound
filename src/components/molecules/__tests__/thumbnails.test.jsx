import React from 'react';
import {create} from 'react-test-renderer';

import Thumbnails from '../thumbnails.jsx';

describe('Thumbnails', () => {
	it('renders correctly with list of zero', () => {
		const edges = [];
		const title = 'title';
		const type = 'type';
		const component = create(<Thumbnails edges={edges} title={title} type={type}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with list of one', () => {
		const picture = {coordinates: {latitude: 52.123, longitude: 13.123}, image: {}, title: 'title1'};
		const thumbnailData = {data: picture, id: 'id-foo', uid: 'uid-bar'};
		const thumbnailDataNode = {node: thumbnailData};
		const edges = [thumbnailDataNode];
		const title = 'title';
		const type = 'type';
		const component = create(<Thumbnails edges={edges} title={title} type={type}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with list of more', () => {
		const picture1 = {coordinates: {latitude: 52.456, longitude: 13.456}, image: {}, title: 'title1'};
		const picture2 = {coordinates: {latitude: 52.789, longitude: 13.798}, image: {}, title: 'title2'};
		const thumbnailData1 = {data: picture1, id: 'id-1', uid: 'uid-1'};
		const thumbnailData2 = {data: picture2, id: 'id-2', uid: 'uid-2'};
		const thumbnailDataNode1 = {node: thumbnailData1};
		const thumbnailDataNode2 = {node: thumbnailData2};
		const edges = [thumbnailDataNode1, thumbnailDataNode2];
		const title = 'title';
		const type = 'type';
		const component = create(<Thumbnails edges={edges} title={title} type={type}/>);
		expect(component).toMatchSnapshot();
	});
});
