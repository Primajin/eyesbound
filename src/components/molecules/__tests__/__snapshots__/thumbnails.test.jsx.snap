// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails renders correctly with list of more 1`] = `
Array [
  <h1>
    type
    : 
    title
     (
    2
    )
  </h1>,
  <ul
    css={
      Object {
        "map": undefined,
        "name": "148b9zh",
        "next": undefined,
        "styles": "
	display: grid;
	grid-template-columns: 1fr 1fr 1fr 1fr;
	gap: 5px;

	@media (min-width: 576px) {
		gap: 10px;
	};

	li {
		content-visibility: auto;
	}
",
        "toString": [Function],
      }
    }
  >
    <li>
      <a
        aria-label="link-to-picture"
        href="/picture/uid-1"
      >
        <picture>
          <img
            alt="title1"
            css={
              Object {
                "map": undefined,
                "name": "1w62pol",
                "next": undefined,
                "styles": "
	height: 174px;
	width: 100%;
",
                "toString": [Function],
              }
            }
            height={174}
            image={
              Object {
                "layout": "CONSTRAINED",
              }
            }
            width={261}
          />
        </picture>
      </a>
    </li>
    <li>
      <a
        aria-label="link-to-picture"
        href="/picture/uid-2"
      >
        <picture>
          <img
            alt="title2"
            css={
              Object {
                "map": undefined,
                "name": "1w62pol",
                "next": undefined,
                "styles": "
	height: 174px;
	width: 100%;
",
                "toString": [Function],
              }
            }
            height={174}
            image={
              Object {
                "layout": "CONSTRAINED",
              }
            }
            width={261}
          />
        </picture>
      </a>
    </li>
  </ul>,
]
`;

exports[`Thumbnails renders correctly with list of one 1`] = `
Array [
  <h1>
    type
    : 
    title
     (
    1
    )
  </h1>,
  <ul
    css={
      Object {
        "map": undefined,
        "name": "148b9zh",
        "next": undefined,
        "styles": "
	display: grid;
	grid-template-columns: 1fr 1fr 1fr 1fr;
	gap: 5px;

	@media (min-width: 576px) {
		gap: 10px;
	};

	li {
		content-visibility: auto;
	}
",
        "toString": [Function],
      }
    }
  >
    <li>
      <a
        aria-label="link-to-picture"
        href="/picture/uid-bar"
      >
        <picture>
          <img
            alt="title1"
            css={
              Object {
                "map": undefined,
                "name": "1w62pol",
                "next": undefined,
                "styles": "
	height: 174px;
	width: 100%;
",
                "toString": [Function],
              }
            }
            height={174}
            image={
              Object {
                "layout": "CONSTRAINED",
              }
            }
            width={261}
          />
        </picture>
      </a>
    </li>
  </ul>,
]
`;

exports[`Thumbnails renders correctly with list of zero 1`] = `
Array [
  <h1>
    type
    : 
    title
     (
    0
    )
  </h1>,
  <ul
    css={
      Object {
        "map": undefined,
        "name": "148b9zh",
        "next": undefined,
        "styles": "
	display: grid;
	grid-template-columns: 1fr 1fr 1fr 1fr;
	gap: 5px;

	@media (min-width: 576px) {
		gap: 10px;
	};

	li {
		content-visibility: auto;
	}
",
        "toString": [Function],
      }
    }
  />,
]
`;
